lib/TPP/TestMatchers.cpp

```diff
@@ -159,6 +159,20 @@ void testTppIdentity(FunctionOpInterface funcOp) {
   });
 }
 
+void testTppRank(FunctionOpInterface funcOp) {
+  // clang-format off
+  auto matcher =
+    StructuredOpMatcher::make<linalg::GenericOp>()
+      .input(MatchAll(), HasRank({HasRank::SCALAR, 1}));
```

Comment:
This is very little testing...

Reply:
Why does `matcher0` have both `SCALAR` and `1`?

Reply:
this is fixed.